### ISSUE TITLE
client-discussion 

### ISSUE THREAD

**Issue by quangogster** [quangogster/todd-app#2]

**Comment by tzusman** [comment]
please commit directly to github so i can track progress

**Comment by quangogster** [comment]
okay. I will make commit more frequent.

**Comment by quangogster** [comment]
Hey @tzusman, how are you doing?  @danielr18 is my colleague and we have worked together on my many projects.  Hope that is okay with you.

**Comment by danielr18** [comment]
Hi @tzusman, I'm Daniel and as Quang said, we work together. If you want to ask something about the components being created, you can tag me in a comment, and I'll comment back asap.

**Comment by tzusman** [comment]
Hey Daniel, good to meet you.  Work looks good so far!

**Comment by quangogster** [comment]
Hey @tzusman, would you like to take a look at the project and let us know what you think?

**Comment by tzusman** [comment]
Looks awesome!  Two things:

- [ ] The [gallery hover state](http://foxythemes.net/preview/products/beagle2/pages-gallery.html)
- [ ] The timeline thing that is in the mock-up I sent [essentially a horizontal version of this](http://foxythemes.net/preview/products/beagle2/pages-timeline.html) but with two circles on the sides and text in the middle

**Comment by danielr18** [comment]
@tzusman About the timeline, do you want it like in the mock-up? Like 60% of ChatWindow's height being chat messages, and the rest the horizontal timeline.

**Comment by tzusman** [comment]
No, so it's describing an event that occurs sometime in the conversation.
There would be messages above and below, and there would be multiple in the
feed

**Comment by danielr18** [comment]
Would it look like this?

Message 
Message 

Circle - event 1 text - circle
Circle - event 2 text - circle

Message

**Comment by tzusman** [comment]
Yep exactly.  Could you make it a minimum width between the circles of like
70% but if the event is bigger than that space, have it expand?

**Comment by danielr18** [comment]
I can set that minimun width, but I think it would be better to expand it vertically if it's bigger, by adding another line and keeping it center aligned (vetically and horizontally)

**Comment by tzusman** [comment]
You're probably right. Good call. That'll be better on mobile definitely

**Comment by danielr18** [comment]
Hey @tzusman, both changes added. Let us know what you think.

**Comment by tzusman** [comment]
The heroku app isn't different, and locally I'm getting this when I run `yarn install` during the react boilerplate dll task:

```
ERROR in dll reactBoilerplateDeps
Module not found: Error: Can't resolve 'react-icons'
```

**Comment by tzusman** [comment]
Odd thing is that I see react-icons in package.json and there's a folder for it in node_modules

**Comment by danielr18** [comment]
Just pushed to heroku, take a look. I've been using npm install instead of yarn. Let me see what's the the issue with Yarn.

**Comment by tzusman** [comment]
Looks perfect.  you did great work.  So this is the main screen for the app, but there are a couple ancillary screens that I'd like coded up as well.  Would you be open to taking on more work?

**Comment by danielr18** [comment]
Awesome, of course.

**Comment by tzusman** [comment]
How loose of design guidelines would you want for the other work?  They're not complicated views - managing a gallery (multi file upload, edit, delete, etc).  A "gallery" of members view.  I can still get you some loose mockups though.  Just wondering if you want something like I already sent or just sketches.

**Comment by danielr18** [comment]
We already have a theme/style defined, so sketches are fine, doesn't have to be anything fancy.

**Comment by tzusman** [comment]
Okay.  Let me scope out the work and then we can get moving on next steps.  Give me a bit to put that together - I have a busy day.

**Comment by tzusman** [comment]
A request: with future development, please put styles in an external file, next to the component - for example:

`app/components/ChatWindow/ChatInput.js` would match `app/components/ChatWindow/ChatInput.styles.js` which would export `Layout, Input, MediaIcon, SendIcon`

Thanks!

**Comment by tzusman** [comment]
Random question: Instead of this
```
import MdPhotoCamera from 'react-icons/lib/md/photo-camera';
import MdSend from 'react-icons/lib/md/send';
```

can't you do the following since webpack has tree-shaking now?
```
import { MdPhotoCamera, MdSend } from 'react-icons';
```

**Comment by danielr18** [comment]
About the question, take a look at this issue:
https://github.com/gorangajic/react-icons/issues/4

**Comment by danielr18** [comment]
And yes, I'll separate the styled component, would be better.

**Comment by danielr18** [comment]
btw Yarn should be working now.

**Comment by tzusman** [comment]
That thread is over a year old and looks to be solved?

**Comment by danielr18** [comment]
![image](https://cloud.githubusercontent.com/assets/9046616/22804820/25640fbc-eef1-11e6-945c-b8d6e8003c02.png)

I get this error when I try to import them that way, so I used that way to import them.

**Comment by tzusman** [comment]
Oh gotcha. Weird.  Yarn install works locally now, thanks!
